### PR TITLE
EA PG Migrate Trivial Collections (Part 1)

### PR DESCRIPTION
**Pull request by oetherington**
Many of our collections are "trivial"; that is, they never join with anything and they're only ever used for simple gets/sets - nothing too fancy. This PR converts all of these collections to Postgres for the EA Forum. The full list is:
 - DatabaseMetadata (EDIT: I'm no longer including this in this PR)
 - AdvisorRequests
 - Bans
 - EmailTokens
 - FeaturedResources
 - GardenCodes
 - LegacyData
 - Migrations
 - PetrovDayLaunchs (yes, that's how it's spelled)
 - RSSFeeds
 - ModerationTemplates
 - PodcastEpisodes
 - Podcasts
 - Reports
 - Spotlights

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203480950183109) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
For DatabaseMetadata, please submit a PR to ForumCredentials before merging this PR

File: packages/lesswrong/lib/collections/advisorRequests/collection.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 import { addUniversalFields, getDefaultResolvers } from '../../collectionUtils'
-import { getDefaultMutations, MutationOptions } from '../../vulcan-core/default_mutations';
+import { getDefaultMutations } from '../../vulcan-core/default_mutations';
+import { forumTypeSetting } from '../../instanceSettings';
 
 export const AdvisorRequests: AdvisorRequestsCollection = createCollection({
   collectionName: 'AdvisorRequests',
   typeName: 'AdvisorRequest',
-  collectionType: 'mongo',
+  collectionType: forumTypeSetting.get() === 'EAForum' ? 'switching' : 'mongo',
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/advisorRequests/collection.ts:10]: As mentioned, we're holding off on this?

File: packages/lesswrong/lib/sql/SelectQuery.ts [hunk 2]
```diff
@@ -148,6 +148,21 @@ class SelectQuery<T extends DbObject> extends Query<T> {
...
   }
 
+  private selectorIsEmpty(selector?: string | MongoSelector<T>): boolean {
+    if (!selector) {
+      return true;
+    }
+
+    if (typeof selector === "string") {
+      return false;
+    }
```
Comment (jpaddison3) [inline packages/lesswrong/lib/sql/SelectQuery.ts:158]: I assume that not checking for length here is an intentional change in behavior?

File: packages/lesswrong/server/serverStartup.ts [hunk 3]
```diff
@@ -111,6 +107,10 @@ async function serverStartup() {
     }
   }
 
+  // eslint-disable-next-line no-console
+  console.log("Running onStartup functions");
+  await runStartupFunctions();
+
```
Comment (jpaddison3) [inline packages/lesswrong/server/serverStartup.ts:113]: Discuss this move?

**Review by jpaddison3 — APPROVED** [review]
ASTLAC

File: packages/lesswrong/lib/sql/SelectQuery.ts (hunk 2 above)
Comment (oetherington) [reply packages/lesswrong/lib/sql/SelectQuery.ts:158]: If selector is "" the we returned true at lines 152/153

File: packages/lesswrong/server/serverStartup.ts (hunk 3 above)
Comment (oetherington) [reply packages/lesswrong/server/serverStartup.ts:113]: It turns out that migrating these collections requires running a startup function that relies on the internal representation of Postgres tables already being built. This just makes sure those tables are initialised before running the startup functions.

**Pull request merged**